lib/Dialect/SV/SVOps.cpp

```diff
@@ -643,21 +643,44 @@ auto CaseOp::getCases() -> SmallVector<CaseInfo, 4> {
   return result;
 }
 
+/// Parse case op.
+/// case op ::= `sv.case` ?case-style ?unique-priority cond attr-dict `:` type
+///              case-pattern^*
+/// case-style ::= `case` | `casex` | `casez`
+/// unique-priority (see SV Spec 12.5.3) ::= `unique` | `unique0` | `priority`
+/// case-pattern ::= `case` bit-pattern `:` region
 ParseResult CaseOp::parse(OpAsmParser &parser, OperationState &result) {
   auto &builder = parser.getBuilder();
 
   OpAsmParser::UnresolvedOperand condOperand;
   Type condType;
+  // Track `unique`, `unique0` or `priority`.
+  llvm::Optional<UniquePriorityType> uniquePriority;
 
   auto loc = parser.getCurrentLocation();
 
   StringRef keyword;
   if (!parser.parseOptionalKeyword(&keyword)) {
     auto kind = symbolizeCaseStmtType(keyword);
-    if (!kind.hasValue())
-      return parser.emitError(loc, "expected 'case', 'casex', or 'casez'");
-    auto caseEnum = static_cast<int32_t>(kind.getValue());
-    result.addAttribute("caseStyle", builder.getI32IntegerAttr(caseEnum));
+    if (kind.hasValue()) {
+      auto caseEnum = static_cast<int32_t>(kind.getValue());
+      result.addAttribute("caseStyle", builder.getI32IntegerAttr(caseEnum));
+    } else {
+      uniquePriority = symbolizeUniquePriorityType(keyword);
+      if (!uniquePriority)
+        return parser.emitError(loc, "expected 'case', 'casex', 'casez', "
+                                     "'unique', 'unique0', or 'priority'");
+    }
+  }
+
+  if (uniquePriority || !parser.parseOptionalKeyword(&keyword)) {
+    if (!uniquePriority)
+      uniquePriority = symbolizeUniquePriorityType(keyword);
+    if (!uniquePriority)
+      return parser.emitError(loc,
+                              "expected 'unique', 'unique0', or 'priority'");
+    auto caseEnum = static_cast<int32_t>(uniquePriority.getValue());
+    result.addAttribute("uniquePriority", builder.getI32IntegerAttr(caseEnum));
```

Comment:
Can this logic be flattened out a bit? The parsing the unique attr in either of the blocks is a bit funky.

Reply:
This mess is because both case style (`case`, `casez`, and `casex`) and validation qualifier are optional to parse. I modified 
case style to be mandatory on IR.

Reply:
On the second thought, I reverted changes related to IR parser because validation qualifier sounds like "meta data" rather a IR component. It would be enough to put the attribute on attr-dict. 

```mlir
sv.case %cond : i1 {validationQualifier = #sv<"validation_qualifier unique">}
```